shared/utility/module_test_utils/ModuleTester.hpp

```diff
@@ -0,0 +1,121 @@
+/*
+ * This file is part of NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2021 NUbots <nubots@nubots.net>
+ */
+
+#ifndef UTILITY_MODULETESTUTILS_MODULETESTER
+#define UTILITY_MODULETESTUTILS_MODULETESTER
+
+#define CATCH_CONFIG_RUNNER
+
+#include <catch.hpp>
+
+#include "TestLogHandler.hpp"
+
+namespace utility::module_test {
+
+    /**
+     * @brief Class for running discrete PowerPlants to test Reactors and Reactions
+     *
+     * @tparam ModuleToTest The module which is to be tested
+     * @tparam timeout_ticks  The number of ticks of the TimeoutTimeUnit before the test times out and fails
+     * @tparam TimeoutTimeUnit The unit of time which the timeout_ticks are
+     */
+    template <typename ModuleToTest, int timeout_ticks = 500, typename TimeoutTimeUnit = std::chrono::milliseconds>
```

Comment:
Do we want to let people chose their own timeout time unit? I'm just thinking that consistency in time units might be good across the tests. The only thing I can think of is if we're testing to make sure the module runs quickly and we want more precision than milliseconds, though I'm not sure that's likely?

Reply:
Without a good reason to do otherwise, I think it's better to either make things have units or use SI units. And since we can't use seconds here, proving the ability to control the units (and in a way documenting what it is if you don't use the units) is good